### PR TITLE
Base fee for cryptoTransfers involving Tokens with CustomFee should be $0.002 

### LINKED ISSUE #1817: (HIP-17) The base price of cryptoTransfer of a Token with customFee should be $0.002 in case of a `FUNGIBLE_COMMON` and $0.002 in case of a `NON_FUNGIBLE_UNIQUE`
### Problem

With possible nested cryptoTransfers, the base price of transferring a token with customFee should be $0.002.

### Solution

Add subTypes `TOKEN_FUNGIBLE_COMMON_WITH_CUSTOM_FEES` and `TOKEN_NON_FUNGIBLE_UNIQUE_WITH_CUSTOM_FEES` to `cryptoTransfer` FeeSchedule to ensure that the base price of transferring a :

- Fungible Common token with customFee is $0.002
- Non Fungible Unique token with customFee is $0.002

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by anighanta**
**Description**:

Charge $0.002 for base fungible common token transfer and non fungible unique token tranfer with customFees in both cases

**Related issue(s)**:

Fixes #1817 

**Notes for reviewer**:

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by tinker-michaelj — APPROVED** [review]
👍 Looks great!! I think CodeCov is complaining about partial coverage of the `if` conditions in `CryptoTransferMeta.getSubType()`, nbd.

**Comment by anighanta** [comment]
> 👍 Looks great!! I think CodeCov is complaining about partial coverage of the `if` conditions in `CryptoTransferMeta.getSubType()`, nbd.

I did cover the if conditions in a unit test in `CryptoTransferMetaTest.getSubTypePrioritizesNFT()`. will ignore codecov for now.

File: hapi-fees/src/main/java/com/hedera/services/pricing/BaseOperationUsage.java [hunk 1]
```diff
@@ -94,36 +94,39 @@
...
 					return uniqueTokenMint();
-				case TokenAccountWipe:
+				}
+			case TokenAccountWipe:
+				if (type == TOKEN_NON_FUNGIBLE_UNIQUE) {
 					return uniqueTokenWipe();
-				case TokenBurn:
+				}
+			case TokenBurn:
+				if (type == TOKEN_NON_FUNGIBLE_UNIQUE) {
```
Comment (qnswirlds) [inline hapi-fees/src/main/java/com/hedera/services/pricing/BaseOperationUsage.java:121]: 3 uncovered conditions
https://sonarcloud.io/component_measures?branch=01817-M-BaseFeeForTokensWithCustomFee-TokenTransfer&id=com.hedera.hashgraph%3Ahedera-services&metric=new_uncovered_conditions&selected=com.hedera.hashgraph%3Ahedera-services%3Ahapi-fees%2Fsrc%2Fmain%2Fjava%2Fcom%2Fhedera%2Fservices%2Fpricing%2FBaseOperationUsage.java&view=list

File: hapi-fees/src/main/java/com/hedera/services/usage/crypto/CryptoTransferMeta.java [hunk 2]
```diff
@@ -85,10 +85,16 @@ public int getNumNftOwnershipChanges() {
...
+			if (customFeeHbarTransfers > 0 || customFeeTokenTransfers > 0) {
+				return SubType.TOKEN_NON_FUNGIBLE_UNIQUE_WITH_CUSTOM_FEES;
+			}
 			return SubType.TOKEN_NON_FUNGIBLE_UNIQUE;
 		}
-		if(numFungibleTokenTransfers != 0) {
+		if (numFungibleTokenTransfers != 0) {
+			if (customFeeHbarTransfers > 0 || customFeeTokenTransfers > 0) {
+				return SubType.TOKEN_FUNGIBLE_COMMON_WITH_CUSTOM_FEES;
+			}
```
Comment (qnswirlds) [inline hapi-fees/src/main/java/com/hedera/services/usage/crypto/CryptoTransferMeta.java:97]: 2 uncovered conditions
https://sonarcloud.io/component_measures?branch=01817-M-BaseFeeForTokensWithCustomFee-TokenTransfer&id=com.hedera.hashgraph%3Ahedera-services&metric=new_uncovered_conditions&selected=com.hedera.hashgraph%3Ahedera-services%3Ahapi-fees%2Fsrc%2Fmain%2Fjava%2Fcom%2Fhedera%2Fservices%2Fusage%2Fcrypto%2FCryptoTransferMeta.java&view=list

**Review by qnswirlds — COMMENTED** [review]
I think these were what Michael meant.

File: test-clients/src/main/java/com/hedera/services/bdd/suites/token/TokenTransactSpecs.java [hunk 3]
```diff
@@ -92,32 +92,32 @@ public static void main(String... args) {
...
+//						uniqueTokenTxnWithNoAssociation(),
+//						uniqueTokenTxnWithFrozenAccount(),
+//						uniqueTokenTxnWithSenderNotSigned(),
+//						uniqueTokenTxnWithReceiverNotSigned(),
+//						uniqueTokenTxnsAreAtomic(),
+//						uniqueTokenDeletedTxn(),
+//						cannotSendFungibleToDissociatedContractsOrAccounts(),
+//						cannotGiveNftsToDissociatedContractsOrAccounts(),
+//						recordsIncludeBothFungibleTokenChangesAndOwnershipChange(),
+//						transferListsEnforceTokenTypeRestrictions(),
```
Comment (ljianghedera) [inline test-clients/src/main/java/com/hedera/services/bdd/suites/token/TokenTransactSpecs.java:116]: Should these and the following 3 test cases be uncommented back?

**Pull request merged**